e/egeria/egeria_ubuntu_20.04.sh

```diff
@@ -0,0 +1,49 @@
+# ----------------------------------------------------------------------------
+#
+# Package       : egeria
+# Version       : egeria-release-2.10 
+# Source repo   : https://github.com/odpi/egeria
+# Tested on     : Ubuntu 20.04.1 LTS (Focal Fossa)
+# Script License: Apache License, Version 2 or later
+# Maintainer    : Nagesh Tarale <Nagesh.Tarale@ibm.com>
+#
+# Disclaimer: This script has been tested in non-root mode on given
+# ==========  platform using the mentioned version of the package.
+#             It may not work as expected with newer versions of the
+#             package and/or distribution. In such case, please
+#             contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+
+#!/bin/bash
+
+export URL=https://github.com/odpi/egeria.git
+export BRANCH="$1"
+PKG_NAME=${URL##*/}
+PKG_NAME=${PKG_NAME%%.*}
+
+if [ -d $PKG_NAME ] ; then
+  rm -rf $PKG_NAME
+fi
+# Verify the arguments passed from command line.
+
+if [ $# -lt 1 ]
+then
+  echo "Usage: $0 branch or tag_name >> For specific versions, default is master"
+  BRANCH="master"
+fi
+#echo "Branch is $1 and no of arguments $# "
+if [ $# -gt 1 ]
+then
+        echo "Specify no of arguments for specific versions by default it will be master, USAGE: $0 BRANCH"
+        exit 0
+else
+        echo "Cloning the target"
+fi
+
+# Build and test the package
+# BRANCHES can be master, or any other tags supported.
+# Ex : egeria-release-2.10 , egeria-release-2.9, master etc
+git clone --depth 1 --branch $BRANCH $URL --single-branch
```

Comment:
What will happen if $1 is invalid argument ? Should it not exit here instead of continuing with build ?

Reply:
"It will FAIL*ubuntu@nagesh-at:/scratch$ ./egeria.sh lalaal
Cloning the target
Cloning into 'egeria'...
warning: Could not find remote branch lalaal to clone.
fatal: Remote branch lalaal not found in upstream origin
./egeria.sh: line 48: cd: egeria: No such file or directory
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 11.0.11, vendor: Ubuntu, runtime: /usr/lib/jvm/java-11-openjdk-ppc64el
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "5.4.0-42-generic", arch: "ppc64le", family: "unix"
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.104 s
[INFO] Finished at: 2021-06-24T06:18:42Z
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/scratch). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException
ubuntu@nagesh-at:/scratch$

Reply:
ubuntu@nagesh-at:/scratch$ ./egeria.sh lalaal
Cloning the target
Cloning into 'egeria'...
warning: Could not find remote branch lalaal to clone.
fatal: Remote branch lalaal not found in upstream origin
./egeria.sh: line 48: cd: egeria: No such file or directory
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 11.0.11, vendor: Ubuntu, runtime: /usr/lib/jvm/java-11-openjdk-ppc64el
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "5.4.0-42-generic", arch: "ppc64le", family: "unix"
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.104 s
[INFO] Finished at: 2021-06-24T06:18:42Z
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/scratch). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException
ubuntu@nagesh-at:/scratch$

Reply:
It will fail with above mentioned message.

Reply:
It is attempting to cd to a non-existent directory, trying a build in a nonstandard directory and failing. This is not the way of error handling